services/spar/test-integration/Test/Spar/Scim/UserSpec.hs

```diff
@@ -73,13 +74,93 @@ spec = do
+      handle'@(Handle handle) <- nextHandle
+      runSpar $ Intra.setBrigUserHandle member handle'
+      [user] <- listUsers tok (Just (filterBy "userName" handle))
+      lift $ (Scim.User.active . Scim.value . Scim.thing $ user) `shouldBe` Just True
+    let activeInactiveAndBack f = do
+          user <- randomScimUser
+          (tok, _) <- registerIdPAndScimToken
+          scimStoredUserBlah <- createUser tok user
+          let uid = Scim.id . Scim.thing $ scimStoredUserBlah
+          do
+            -- NOTE: It's Nothing, not Just True, as we want to also test the fact that existing
+            -- SCIM records don't have the active field. absence of active should be interpreted as Active.
+            -- Once we get rid of the `scim` table and make scim serve brig records directly, this is
+            -- not an issue anymore.
+            lift $ (Scim.User.active . Scim.value . Scim.thing $ scimStoredUserBlah) `shouldBe` Nothing
+            void $ aFewTimes (runSpar $ Intra.getStatus uid) (== Active)
+          do
+            scimStoredUser <- f tok uid user True
+            lift $ (Scim.User.active . Scim.value . Scim.thing $ scimStoredUser) `shouldBe` Just True
+            void $ aFewTimes (runSpar $ Intra.getStatus uid) (== Active)
+          do
+            scimStoredUser <- f tok uid user False
+            lift $ (Scim.User.active . Scim.value . Scim.thing $ scimStoredUser) `shouldBe` Just False
+            void $ aFewTimes (runSpar $ Intra.getStatus uid) (== Suspended)
+          do
+            scimStoredUser <- f tok uid user True
+            lift $ (Scim.User.active . Scim.value . Scim.thing $ scimStoredUser) `shouldBe` Just True
+            void $ aFewTimes (runSpar $ Intra.getStatus uid) (== Active)
+    it "PUT will change state from active to inactive and back" $ do
+      void $ activeInactiveAndBack $ \tok uid user active ->
+        updateUser tok uid user {Scim.User.active = Just active}
+    it "PATCH will change state from active to inactive and back" $ do
+      let replaceAttrib name value =
+            PatchOp.Operation
+              PatchOp.Replace
+              (Just (PatchOp.NormalPath (Filter.topLevelAttrPath name)))
+              (Just (toJSON value))
+      void $ activeInactiveAndBack $ \tok uid _user active ->
+        patchUser tok uid $ PatchOp.PatchOp [replaceAttrib "active" active]
+    -- NOTE: This one might be wrong? Reviewer should carefully decide if they agree.
+    -- Rationale:  when we GET data from scim currently, it doesnt check
+    -- at all if it is consistent with brig at the moment. it just returns what we have.
+    -- We make the assumption that when the `active` field is not there, that the user is
+    -- active by default; such that the IdP will de-activate it when it disagrees with the IdP
+    -- state.   Now if the IdP wants to remove the "active" field altogether, we dont "know"
+    -- anything again, and this should cause the user to be "active" again. This
```

Comment:
"This" is not a complete sentence :-)  It's also not clear how an IdP comes into play here.  Do you mean the SCIM peer that patches the user?  What does it mean if "it" (the IdP, presumably) disagrees with the (same?) IdP state?

Anyway I think it makes no sense to change the activation state if the SCIM peer explicitly does *NOT* patch it.  I think we should leave it at whatever it is in that case.  I suspect the reason why you came up with the current semantics is due to the confusion about the different representations of the user in brig and spar.  See https://github.com/zinfra/backend-issues/issues/1006

Reply:
4a1fe43c0c961e28e4709e433e3d4216dcf3400b 28ae745ce136fa7ade54a4a9d863c3c94111d40b